runtime/drivers/clickhouse/olap.go

```diff
@@ -251,7 +251,70 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 		})
 	}
 
-	// merge strategy is also not supported for clickhouse
+	if strategy == drivers.IncrementalStrategyMerge {
+		_, onCluster, err := informationSchema{c: c}.entityType(ctx, "", name)
+		if err != nil {
+			return err
+		}
+		onClusterClause := ""
+		if onCluster {
+			onClusterClause = "ON CLUSTER " + safeSQLName(c.config.Cluster)
+		}
+		// Get the engine info of the given table
+		engine, err := c.getTableEngine(ctx, name)
+		if err != nil {
+			return err
+		}
+		// Distributed table cannot be altered directly, so we need to alter the local table
+		if engine == "Distributed" {
+			name = localTableName(name)
+		}
+		// create temp table with the same schema
+		tempName := tempName(name)
+		err = c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("CREATE OR REPLACE TABLE %s %s AS %s", safeSQLName(tempName), onClusterClause, name),
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		// clean up the temp table
+		defer func() {
+			err = c.Exec(ctx, &drivers.Statement{
+				Query:    fmt.Sprintf("DROP TABLE %s %s", safeSQLName(tempName), onClusterClause),
```

Comment:
There is a risk that the `ctx` is cancelled (e.g. if the pod is restarted or it hits the reconciliation timeout). Since `tempName` function returns a random table name, this means the table will never be cleaned up.

Imagine if this operation is too slow for the current resources and keeps getting cancelled/retried. Then a lot of temp tables can accumulate.

Have you considered one or both of these options:
1. If `ctx.Err() != nil`, then creating a new short-lived background context to provide a brief opportunity for cleanup (e.g. `context.Background().WithTimeout(15*time.Second)`).
2. Changing `tempName` to return a deterministic temporary table name (not sure, but could be e.g. `__rill_temp_{table name}_{sql hash}`). This would ensure there's at most one orphaned temp table per table/split.